addons/sale_loyalty_delivery/tests/test_free_shipping_reward.py

```diff
@@ -400,3 +400,33 @@ def test_delivery_shant_count_toward_quantity_bought(self):
         # Make sure the promotion is NOT added
         err_msg = "No reward lines should be created as the delivery line shouldn't be included in the promotion calculation"
         self.assertEqual(len(order.order_line.ids), 2, err_msg)
+
+    def test_free_shipping(self):
+        """ Test that a free shipping coupon is correctly applied. """
+        program = self.env['loyalty.program'].create({
+            'name': 'Free Shipping',
+            'trigger': 'auto',
+            'rule_ids': [(0, 0, {})],
+            'reward_ids': [(0, 0, {
+                'reward_type': 'shipping',
+            })],
+        })
+        order = self.env['sale.order'].create({
+            'partner_id': self.steve.id,
+            'order_line': [
+                (0, False, {
+                    'product_id': self.product_B.id,
+                    'name': 'Product B',
+                    'product_uom': self.uom_unit.id,
+                    'product_uom_qty': 1.0,
+                })
+            ]
+        })
+        self._auto_rewards(order, program)
+        delivery_wizard = Form(self.env['choose.delivery.carrier'].with_context({
+            'default_order_id': order.id,
+            'default_carrier_id': self.env['delivery.carrier'].search([])[1]
+        }))
+        choose_delivery_carrier = delivery_wizard.save()
+        choose_delivery_carrier.button_confirm()
+        self.assertEqual(choose_delivery_carrier.delivery_price, 0.0)
```

Comment:
I know there is a bit of mismatch between `loyalty_delivery` (with delivery depending on `sale`) and `sale_delivery_loyalty`, it's fixed in 17.0, but it's strange to see this test in another module than the fix. If you need some code from `sale_delivery_loyalty`, put your fix in that module too. Both will always be installed together anyway.